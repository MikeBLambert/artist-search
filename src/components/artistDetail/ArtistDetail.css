#artistDetail {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: lightgrey;
  margin-top: 45px;
  border-radius: 15px;
  border: 5px solid black;
  box-shadow: 2px 2px lightgrey;

  #title {
    color: black;
    font-family: 'Black Ops One', cursive;
    font-size: 50px;
    margin-top: 60px;
  }
}
