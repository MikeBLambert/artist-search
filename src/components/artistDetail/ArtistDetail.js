import React, { Component, Fragment } from 'react';
import { getArtist } from '../../services/artistApi';
import Songs from './Songs';
// import PropTypes from 'prop-types';

export default class ArtistDetail extends Component {
  state = {
    artist: '',
    songs: [],
    loading: true
  };

  fetchSongs = () => {
    const id = this.props.location.pathname.slice(1).split('/')[1];
    getArtist(id)
      .then(({ artist, songs }) => {
        this.setState({ artist, songs, loading: false });
      });
  };

  componentDidMount() {
    this.fetchSongs();
  }

  render() {

    const { artist, songs, loading } = this.state;
    const songsComponent = !loading && <Songs songs={songs} artist={artist} />;

    return (
      <Fragment>
        {songsComponent}
      </Fragment>
    );
  }

}
