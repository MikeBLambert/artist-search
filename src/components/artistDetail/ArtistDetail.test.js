import React from 'react';
import ArtistDetail from './ArtistDetail';
import { shallow } from 'enzyme';

describe('ArtistDetail', () => {
  it('renders an ArtistDetail component', () => {
    const wrapper = shallow(<ArtistDetail />);
    expect(wrapper).toMatchSnapshot();
  });

  it('has a songs component', () => {
    const wrapper = shallow(<ArtistDetail />);
    expect(wrapper.find('Songs')).toHaveLength(1);
  });
});
