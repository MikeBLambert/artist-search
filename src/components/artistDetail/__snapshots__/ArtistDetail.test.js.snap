// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistDetail renders an ArtistDetail component 1`] = `
<Fragment>
  <div
    id="artistDetail"
  >
    <h1
      id="artistTitle"
    >
      Songs by 
    </h1>
  </div>
</Fragment>
`;
